src/app/desktop/windows/terminal/terminal-states.ts

```diff
@@ -39,27 +39,25 @@ export abstract class CommandTerminalState implements TerminalState {
 
   protocol: string[] = [];
 
-  executeCommand(command: string, args: string[]) {
-    command = command.toLowerCase();
+  executeCommand(commandToExecute: string, args: string[]) {
+    let command: string = commandToExecute.toLowerCase();
+
     if (this.commands.hasOwnProperty(command)) {
       this.commands[command].executor(args);
 
-      // Only successful and not-flagged commands will be shown in the protocol.
-      if (!this.commands[command].hideFromProtocol) {
-        if (args.length>0) {
-          this.protocol.unshift(command + ' ' + args.join(' '));
-        } else {
-          this.protocol.unshift(command);
-        }
+      // Flagged commands won't be logged.
+      if (this.commands[command].hideFromProtocol) {
+        return;
       }
-    } else if (command !== '') {
+    } else {
       this.commandNotFound(command);
     }
+    this.protocol.unshift((command + ' ' + args.join(' ')).trim());
```

Comment:
```suggestion
    this.protocol.unshift(`${command} ${args.join(' ')`);
```

Reply:
This way, a command without args would be saved as `'command '`. Thus, the `.trim()` is needed